pkg/iter/entry_iterator.go

```diff
@@ -308,35 +309,143 @@ func (i *heapIterator) Close() error {
 	return nil
 }
 
-func (i *heapIterator) Peek() time.Time {
+func (i *mergeEntryIterator) Peek() time.Time {
 	i.prefetch()
 
 	return i.heap.Peek().Entry().Timestamp
 }
 
 // Len returns the number of inner iterators on the heap, still having entries
-func (i *heapIterator) Len() int {
+func (i *mergeEntryIterator) Len() int {
 	i.prefetch()
 
 	return i.heap.Len()
 }
 
+type entrySortIterator struct {
+	heap interface {
+		heap.Interface
+		Peek() EntryIterator
+	}
```

Comment:
I'm curious why we are using a heap. Did we compare with sorting `is []EntryIterator` once? As long as there are no push calls it should be faster.

Reply:
I went for a quick and safe algo extraction it's possible we can now find something faster since they have clear separate purpose but did not investigate more.

This is the point of this PR, we can improve later, I'm sure it's possible. I wanted something that works first as a replacement.

However I don't think that true that you need to order once. IMO you need to reorder everytime you advance an iterator. From the list.

Reply:
The heap helps me keep an ordered state.

Specially when I advance the iterator I call fox to reorder the heap and keep peeking at the top

Reply:
> I went for a quick and safe algo extraction it's possible we can now find something faster since they have clear separate purpose but did not investigate more.

Got it.

> IMO you need to reorder everytime you advance an iterator. From the list.

Ah, so the iterators may overlap. I didn't know. I'm still curious. If the list is sorted and only the first element is changed by popping the iterator it's also the only element that needs to be pushed back. Maybe bubble sort or something like timsort which is fast for mostly sorted data might be faster.